src/server/grpc/p2pool.rs

```diff
@@ -121,17 +123,28 @@ where S: ShareChain
             list_of_templates_sha3x: RwLock::new(VecDeque::with_capacity(MAX_STORED_TEMPLATES_SHA3X + 1)),
             template_store_rx: RwLock::new(HashMap::new()),
             list_of_templates_rx: RwLock::new(VecDeque::with_capacity(MAX_STORED_TEMPLATES_RX + 1)),
-            are_we_synced_with_p2pool,
+            are_we_synced_with_randomx_p2pool,
+            are_we_synced_with_sha3x_p2pool,
         })
     }
 
     /// Submits a new block to share chain and broadcasts to the p2p network.
     pub async fn submit_share_chain_block(&self, block: Arc<P2Block>) -> Result<(), Status> {
-        if !self.are_we_synced_with_p2pool.load(Ordering::Relaxed) {
-            info!(target: LOG_TARGET, "We are not synced yet, not submitting block atm");
-            return Ok(());
-        }
         let pow_algo = block.original_header.pow.pow_algo;
+        match pow_algo {
+            PowAlgorithm::RandomX => {
+                if !self.are_we_synced_with_randomx_p2pool.load(Ordering::Relaxed) {
```

Comment:
Use the most strict one, I think it's SeqCt or something

```suggestion
                if !self.are_we_synced_with_randomx_p2pool.load(Ordering::Seq) {
```